airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py

```diff
@@ -114,8 +116,8 @@ async def publish(
                 spec_cache_bucket_name=spec_cache_bucket_name,
                 metadata_service_gcs_credentials=metadata_service_gcs_credentials,
                 metadata_bucket_name=metadata_service_bucket_name,
-                docker_hub_username=ctx.obj["docker_hub_username"],
-                docker_hub_password=ctx.obj["docker_hub_password"],
+                docker_hub_username=Secret("docker_hub_username", ctx.obj["secret_stores"]["in_memory"]),
+                docker_hub_password=Secret("docker_hub_password", ctx.obj["secret_stores"]["in_memory"]),
```

Comment:
Because the `PublishContext` requires these two parameters to not be `None`.
`ctx.obj["python_registry_token"]` can be set to `None` and the `PublishContext` does handle that.

I believe we would rework the typing, but I did not want to fall in a refactoring rabbit hole too far.